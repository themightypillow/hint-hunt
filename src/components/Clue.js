import React from "react";

const Clue = (props) => (
  <div className="clue">
    {props.text}
    <div className="clue_boxes">
      {
        props.answers.map((answer) => 
        <div key={answer} data-answer={answer} className="clue_checkbox"></div> )
      }
    </div>
  </div>
);

export default Clue;