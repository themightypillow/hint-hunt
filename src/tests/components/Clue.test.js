import React from "react";
import { shallow } from "enzyme";
import Clue from "../../components/Clue";

test("should render Clue correctly", () => {
  const wrapper = shallow(<Clue answers={["A"]} />);
  expect(wrapper).toMatchSnapshot();
});