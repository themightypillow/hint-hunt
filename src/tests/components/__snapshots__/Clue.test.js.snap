// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Clue correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Clue
    answers={
      Array [
        "A",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="clue_boxes"
        >
          <div
            className="clue_checkbox"
            data-answer="A"
          />
        </div>,
      ],
      "className": "clue",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="clue_checkbox"
              data-answer="A"
            />,
          ],
          "className": "clue_boxes",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "A",
            "nodeType": "host",
            "props": Object {
              "className": "clue_checkbox",
              "data-answer": "A",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="clue_boxes"
          >
            <div
              className="clue_checkbox"
              data-answer="A"
            />
          </div>,
        ],
        "className": "clue",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="clue_checkbox"
                data-answer="A"
              />,
            ],
            "className": "clue_boxes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "A",
              "nodeType": "host",
              "props": Object {
                "className": "clue_checkbox",
                "data-answer": "A",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
